components/post_view/post_attachment_container/post_attachment_container.tsx

```diff
@@ -2,16 +2,43 @@
 // See LICENSE.txt for license information.
 
 import React, {useCallback} from 'react';
-import {useHistory} from 'react-router-dom';
+
+import {matchPath, useHistory} from 'react-router-dom';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {isTeamSameWithCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {focusPost} from 'components/permalink_view/actions';
+import {GlobalState} from 'types/store';
 
 export type Props = {
     className?: string;
     children?: JSX.Element;
     link: string;
 };
+
+type LinkParams = {
+    postId: string;
+    teamName: string;
+}
+
+const getTeamAndPostIdFromLink = (link: string) => {
+    const match = matchPath<LinkParams>(link, {path: '/:teamName/pl/:postId'});
+    return match!.params;
```

Comment:
maybe this will always match, but since we already guard against `params` below, would it make sense to change this to:
 
```suggestion
    return match?.params;
```

Reply:
@koox00 Thanks for the suggestion.

I had used  a `non-null assertion operator` to avoid the type error.
```ts
//params.teamName should be a string type, not undefined.
 const shouldFocusPostWithoutRedirect = useSelector((state: GlobalState) => isTeamSameWithCurrentTeam(state, params.teamName));
```
I changed this line to 
```ts
const shouldFocusPostWithoutRedirect = useSelector((state: GlobalState) => isTeamSameWithCurrentTeam(state, params?.teamName ?? ''));
```